debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -322,24 +327,33 @@ private boolean isASignalEventToProcess(T dataCollectionId, Operation operation)
             return handled;
         }
         catch (Exception e) {
-            switch (connectorConfig.getEventProcessingFailureHandlingMode()) {
-                case FAIL:
-                    throw new ConnectException("Error while processing event at offset " + changeRecordEmitter.getOffset().getOffset(), e);
-                case WARN:
-                    LOGGER.warn(
-                            "Error while processing event at offset {}",
-                            changeRecordEmitter.getOffset().getOffset(), e);
-                    break;
-                case SKIP:
-                    LOGGER.debug(
-                            "Error while processing event at offset {}",
-                            changeRecordEmitter.getOffset().getOffset(), e);
-                    break;
-            }
+            handleEventProcessingFailure(e, changeRecordEmitter);
```

Comment:
I'd pass just the offset as parameter instead of changeRecordEmitter. WDYT?

Reply:
I was thinking that changeRecordEmitter might be useful if any additional actions or verifications are needed. But yes, you're right—let's stick to the KISS principle